### DISCUSSION TITLE
How can I send broadcast messages through edge functions?

### DISCUSSION THREAD

**Discussion by wendelfreitas** [supabase/supabase#17124] [Questions]
### Update: [you can broadcast messages through edge functions via our newly released Realtime Broadcast REST API](https://github.com/orgs/supabase/discussions/17124#discussioncomment-7192796)

---

Hello supabase community, I was trying to connect in a specific channel using realtime broadcast but I always got a CHANNEL_ERROR but im not sure if it is possible.

There's a way to send messages to a channel when a edge functions is called ?

Thanks :)

**Comment by laktek** [comment]
@wendelfreitas You can use the newly released Realtime broadcast REST API within Edge Functions to send a message to a channel. ([docs](https://supabase.com/docs/guides/realtime/broadcast#send-messages-using-rest-calls))

Here's an sample edge function that sends a message to a channel called `room-1`.

```typescript
import { createClient } from "https://esm.sh/@supabase/supabase-js@2.38.0";

const SUPABASE_URL = "https://{project_ref}.supabase.co/";
const SUPABASE_KEY = "YOUR_SUPABASE_KEY";

const client = createClient(SUPABASE_URL, SUPABASE_KEY);

Deno.serve(async (req) => {
  const url = new URL(req.url);
  const message = url.searchParams.get("msg");
  const channel = client.channel("room-1");

  // No need to subscribe to channel
  channel
    .send({
      type: "broadcast",
      event: "test",
      payload: { message },
    })
    .then((resp) => {
      console.log(resp);
    });

  // Remember to clean up the channel
  client.removeChannel(channel);

  return new Response("ok");
});
```

Then on the client, if you subscribe to messages on `room-1` channel, you should be able to receive it.

**Reply by skinkairewalker** [reply]
Sugoi  Sugoi

**Reply by rajatsangrame** [reply]
@laktek Can I perform `realtime.broadcast_changes()` inside the edge function?

The idea is, db changes trigger will invoke the edge function, I will perform some operation (modify the payload) and do the broadcast

```
 perform realtime.broadcast_changes(
	    'topic:' || new.id::text,   -- topic
		   tg_op,                          -- event
		   tg_op,                          -- operation
		   tg_table_name,                  -- table
		   tg_table_schema,                -- schema
		   new,                            -- new record
		   old                             -- old record
		);
```
Ref: https://supabase.com/blog/realtime-broadcast-from-database

**Reply by inno-odyssean** [reply]
> @laktek Can I perform `realtime.broadcast_changes()` inside the edge function?
> 
> The idea is, db changes trigger will invoke the edge function, I will perform some operation (modify the payload) and do the broadcast
> 
> ```
>  perform realtime.broadcast_changes(
> 	    'topic:' || new.id::text,   -- topic
> 		   tg_op,                          -- event
> 		   tg_op,                          -- operation
> 		   tg_table_name,                  -- table
> 		   tg_table_schema,                -- schema
> 		   new,                            -- new record
> 		   old                             -- old record
> 		);
> ```
> 
> Ref: https://supabase.com/blog/realtime-broadcast-from-database

any luck on this?

**Reply by GaryAustin1** [reply]
Why don't you just use realtime.send() to send the custom payload in exactly the format you want from your trigger function?
